src/freenas/usr/local/lib/middlewared_truenas/plugins/failover_/event.py

```diff
@@ -525,6 +525,7 @@ def carp_master(self, fobj, ifname, vhid, event, user_override, forcetakeover):
                 self.run_call('etc.generate', 'ssl')
                 # Now we restart the appropriate services to ensure it's using correct certs.
                 self.run_call('service.restart', 'http')
+                self.run_call('service.start', 'mdns', {'ha_propagate': False})
```

Comment:
I know it's impossible to see by looking at the code, but can we move this call below the "critical" services. Right above the `for i in (smartd, ...)` line would make sense.